harpoond.c

```diff
@@ -275,12 +293,21 @@ int main()
     /*
      * KEEP ALIVE AND EVENT LOOP
      */
-    while (RUNNING) {
-        keep_alive(&device);
-        r = libusb_handle_events_timeout_completed(NULL, &zero_tv, NULL);
-        if (r < 0)
-            fprintf(stderr, "libusb failed to handle events: %s\n", libusb_error_name(r));
-        sleep(2);
+    if(RUNNING) {
+        // Force applying new config
+        while (!device.initialized) {
+            keep_alive(&device);
+            r = handle_events(r);
+        }
+
+        for (int i = 0; RUNNING; ++i) {
+            /* Send keep_alive signal every WAIT_TIME s, without blocking whole function. */
+            if(i >= WAIT_TIME / 2) {
```

Comment:
I believe that due to losing packets or some race condition, I tried using 30 but it just would reset after a minute, instead of persisting current settings